markdown/developers/tutorials/adding_modules_to_pipelines.md

```diff
@@ -0,0 +1,49 @@
+---
+title: 'Tutorial: Adding Modules to Pipelines'
+subtitle: Quick reference of the steps needed to add a new module to a pipeline.
+---
+
+This tutorial is a quick reference overview of the steps you will normally take when adding a new module to a pipeline.
+
+Different pipelines may have different workflows, but the following steps will cover the most common aspects.
+
+1. Install module to the pipeline
+
+   ```bash
+   nf-core modules install <tool>/<subtool>
+   ```
+
+2. Add `include` statement at the top of the pipeline's (sub)workflow to import module (as suggested by `nf-core modules install`)
+3. Insert module execution into the relevant place (sub)workflow script `TOOL(ch_input)`
+   - Make sure to mix in the module's version into the version channel, e.g. `ch_versions = ch_versions.mix(TOOL.out.versions)`
+   - Make sure to mix any output files for MultiQC into a relevant channel, e.g. `ch_multiqc_files = ch_multiqc.mix(TOOL.out.log)`
+4. Create a section in `conf/modules.conf` for the module (with a default results directory and output file pattern)
+5. Add any necessary parameters for the module with defaults to `nextflow.config`
+6. Insert the parameters into `ext.args` of corresponding `conf/modules.config` (and in some cases in the module reference in the workflow script)
```

Comment:
Can you clarify this please?

```
and in some cases in the module reference in the workflow script)
```

Reply:
Will do, for reference: `val` inputs to modules e.g. fastp:https://github.com/nf-core/modules/blob/0c078028e29d84f75853552af414019fc5496bab/modules/nf-core/fastp/main.nf#L14